app/Models/WaitingRoom.php

```diff
@@ -29,6 +29,19 @@ public function users()
         return $this->belongsToMany(User::class);
     }
 
+    /**
+     * Determines if the waiting room is open for signupsOpen
+     *
+     * @return bool whether the room is open or not
+     */
+    public function isOpen()
+    {
+        $start = Carbon::parse($this->signup_start_date)->startOfDay();
```

Comment:
These can both be casted to Carbon objects by the model, so you can do `$this->signup_start_date->startOfDay()`.